components/LeagueEntries/LeagueEntries.tsx

```diff
@@ -16,67 +15,112 @@ import Link from 'next/link';
  * @param props.isEliminated - If true, the user is flagged as eliminat4ed
  * @param props.isPickSet - if true, the team logo of the picked team shows up on the LeagueEntries card and the button changes from "make a pick" to "chagne pick"
  * @param props.teamLogo - the team logo
+ * @param props.lockout - if true, the user is locked out from making a pick
  * @returns {React.JSX.Element} - A div element that contains the user's entry information
  */
+
+/**
+ * Display all entries for a league.
+ * @param {string} leagueId - The league id.
+ * @returns {JSX.Element} The rendered entries component.
+ */
 const LeagueEntries = ({
   entryName,
-  linkUrl,
   isEliminated = false,
   isPickSet = false,
+  linkUrl,
+  isLockedOutProp = false,
   teamLogo = '',
-}: ILeagueEntriesProps): JSX.Element => (
-  <div
-    data-testid="league-entry-container-card"
-    className={cn(
-      'league-entry-container-card grid h-20 min-w-fit grid-cols-2 justify-between rounded-lg border border-border bg-card p-4 text-card-foreground shadow-sm',
-      isEliminated ? 'bg-muted' : 'transparent',
-    )}
-  >
-    <section
-      className="league-entry-header flex items-center gap-12"
-      data-testid="league-entry-header"
+}: ILeagueEntriesProps): JSX.Element => {
+  const [isLockedOut, setLockedOut] = useState<boolean>(isLockedOutProp);
+
+  useEffect(() => {
+    /**
+     * Checks if the user is locked out from making a pick
+     */
+    const checkLockout = (): void => {
+      const now = new Date();
+      const day = now.getUTCDay();
+      const hours = now.getUTCHours();
+      if (
+        (day === 4 && hours >= 0) ||
+        (day > 4 && day < 2) ||
+        (day === 2 && hours < 12)
+      ) {
+        setLockedOut(true);
+      } else {
+        setLockedOut(false);
+      }
+    };
+
+    checkLockout();
+
+    const intervalId = setInterval(checkLockout, 60 * 60 * 1000);
+
+    return (): void => clearInterval(intervalId);
```

Comment:
Are you using this `return` in the `useEffect` as cleanup?

Reply:
That is correct. On unMount it ensures it does not continue running and eating up resources, thus hindering performance.